lib/matplotlib/ticker.py

```diff
@@ -377,19 +380,39 @@ class FuncFormatter(Formatter):
     """
     Use a user-defined function for formatting.
 
-    The function should take in two inputs (a tick value ``x`` and a
-    position ``pos``), and return a string containing the corresponding
-    tick label.
+    The function can take in at most two inputs (a required tick value ``x``
+     and an optional position ``pos``), and must return a string containing
+     the corresponding tick label.
     """
     def __init__(self, func):
         self.func = func
 
+        if not isinstance(func, types.BuiltinFunctionType):
+            self.nargs = len(inspect.signature(func).parameters)
+        elif (isinstance(getattr(func, "__self__"), str) and
```

Comment:
Why is this needed at all?  `'{x}'.format(x='eggs', bar='spam')` works fine, so you can just always pass both arguments.

Reply:
`inspect.signature` crashes on builtins, so I need to special case .format (used in lots of the polar examples, my guess is still floating in the wild too.) You're right that I probably don't need to count the arguments sent to format, but since FuncFormat only operates on two it's probably worth letting users know that. Should it be a warning instead?